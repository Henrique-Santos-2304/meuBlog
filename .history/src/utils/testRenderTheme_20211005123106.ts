import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
