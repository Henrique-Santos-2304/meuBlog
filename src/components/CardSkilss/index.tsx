import { Flex, Image, Text } from "@chakra-ui/react";

type cardsFrameworks = {
  url: string;
  title: string;
};
const CardSkilss = ({ url, title }: cardsFrameworks) => (
  <Flex
    h="11.5rem"
    m="4rem 1rem"
    direction="column"
    align="center"
    justify="center"
    borderRadius="1rem"
    bg="whiteAlpha.200"
    transition="transform .4s linear"
    _hover={{ transform: "translate(-7%, -7%)" }}
    isCentered
    cursor="pointer"
  >
    <Image
      src={url}
      alt={title}
      w="80%"
      h="7rem"
      m="0 auto"
      borderRadius="2rem"
      p="1rem"
    />
    <Text
      textAlign="center"
      mt="0.2rem"
      p="0 0.4rem"
      color="gray.100"
      fontWeight="500"
      lineHeight="1.6rem"
      fontSize={["1.1rem", "1.2rem", "1.4rem"]}
    >
      {title}
    </Text>
  </Flex>
);

export default CardSkilss;
