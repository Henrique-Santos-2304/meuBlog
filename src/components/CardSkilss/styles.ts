import styled from "styled-components";
export const Identity = styled.div``;
