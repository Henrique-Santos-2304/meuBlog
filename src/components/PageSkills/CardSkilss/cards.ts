export const image = [
  {
    url: "https://res.cloudinary.com/defnibbpl/image/upload/v1634416786/1024px-Python-logo-notext_jsr03y.png",
    title: "Python",
  },
  {
    url: "https://res.cloudinary.com/defnibbpl/image/upload/v1634417394/javascript-icon_zaobje.svg",
    title: "JavaScript",
  },
  {
    url: "https://res.cloudinary.com/defnibbpl/image/upload/v1634416790/typescript_c5xdpy.svg",
    title: "Typescript",
  },
  {
    url: "https://res.cloudinary.com/defnibbpl/image/upload/v1634417749/w3_html5-icon_sviivv.svg",
    title: "HTML",
  },

  {
    url: "https://res.cloudinary.com/defnibbpl/image/upload/v1634416789/react_eugvgw.svg",
    title: "React",
  },
  {
    url: "https://res.cloudinary.com/defnibbpl/image/upload/v1634416788/nextjs_jbhm7d.svg",
    title: "Next.Js",
  },
  {
    url: "https://res.cloudinary.com/defnibbpl/image/upload/v1634416787/jest_yvhqf3.svg",
    title: "Jest",
  },
  {
    url: "https://res.cloudinary.com/defnibbpl/image/upload/v1634416786/cypress_bmhnah.svg",
    title: "Cypress",
  },
  {
    url: "https://res.cloudinary.com/defnibbpl/image/upload/v1634416789/rtl_i3scd8.svg",
    title: "Testing Library",
  },
  {
    url: "https://res.cloudinary.com/defnibbpl/image/upload/v1634416789/storybook_q1s5cl.svg",
    title: "Storybook",
  },
];
export const stylesImg = [
  {
    url: "https://res.cloudinary.com/defnibbpl/image/upload/v1634417550/styled-components-1_kdt29i.svg",
    title: "Styled Components",
  },
  {
    url: "https://res.cloudinary.com/defnibbpl/image/upload/v1634416786/Bootstrap_logo_utfaxr.svg",
    title: "Bootstrap",
  },
  {
    url: "https://res.cloudinary.com/defnibbpl/image/upload/v1634416786/54212428_dpejja.png",
    title: "Chakra-Ui",
  },
  {
    url: "https://res.cloudinary.com/defnibbpl/image/upload/v1634417414/tailwindcss-icon_1_xozsfd.svg",
    title: "Tailwind-Css",
  },
  {
    url: "https://res.cloudinary.com/defnibbpl/image/upload/v1634417704/w3_css-icon_ijitjf.svg",
    title: "Css",
  },
  {
    url: "https://res.cloudinary.com/defnibbpl/image/upload/v1634416790/strapi_e1wlna.svg",
    title: "Strapi",
  },

  {
    url: "https://res.cloudinary.com/defnibbpl/image/upload/v1634416786/apollo_qhtdll.svg",
    title: "Apolo",
  },
  {
    url: "https://res.cloudinary.com/defnibbpl/image/upload/v1634416786/graphql_c3ix6o.svg",
    title: "GraphQl",
  },
];
