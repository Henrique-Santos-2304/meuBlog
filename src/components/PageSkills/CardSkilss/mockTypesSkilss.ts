export const mockTypesSkillss = {
  url: "img.img.png",
  title: "One Teste",
};
