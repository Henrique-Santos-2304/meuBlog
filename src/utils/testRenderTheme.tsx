import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
