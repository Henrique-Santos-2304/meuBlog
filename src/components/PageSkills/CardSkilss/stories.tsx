import { Story, Meta } from "@storybook/react";
import CardSkilss from ".";

export default {
  title: "CardSkilss",
  component: CardSkilss,
} as Meta;

export const Basic: Story = (args) => <CardSkilss {...args} />;

Basic.args = {
  title: "title basic",
  description: "description basic",
};

export const Default: Story = (args) => <CardSkilss {...args} />;
