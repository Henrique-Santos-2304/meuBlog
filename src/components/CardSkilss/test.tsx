import CardSkilss from ".";
import { renderTheme } from "utils/testRenderTheme";

describe("<CardSkilss />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<CardSkilss />);
    expect(container).toMatchSnapshot();
  });
});
