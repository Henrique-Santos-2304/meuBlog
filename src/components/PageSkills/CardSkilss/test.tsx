import CardSkilss from ".";
import { renderTheme } from "utils/testRenderTheme";
import { mockTypesSkillss } from "./mockTypesSkilss";

describe("<CardSkilss />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<CardSkilss {...mockTypesSkillss} />);
    expect(container).toMatchSnapshot();
  });
});
