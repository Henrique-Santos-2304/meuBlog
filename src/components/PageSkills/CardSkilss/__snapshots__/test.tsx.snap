// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardSkilss /> should render  1`] = `
<div>
   
  <div
    class="css-x5fiqn"
  >
    <img
      alt="One Teste"
      class="chakra-image css-1xbzop4"
      src="img.img.png"
    />
    <p
      class="chakra-text css-18unoc9"
    >
      One Teste
    </p>
  </div>
</div>
`;
